corehq/apps/app_manager/models.py

```diff
@@ -2989,6 +2991,108 @@ def update_app_case_meta(self, app_case_meta):
                 meta.add_closer(self.unique_id, action.close_condition)
 
 
+class ShadowForm(AdvancedForm):
+    form_type = 'shadow_form'
+    # The unqiue id of the form we are shadowing
+    shadow_parent_form_id = StringProperty(required=False, default=None)
+
+    # form actions to be merged with the parent actions
+    extra_actions = SchemaProperty(AdvancedFormActions)
+
+    @property
+    def shadow_parent_form(self):
+        if not self.shadow_parent_form_id:
+            return None
+        else:
+            app = self.get_app()
```

Comment:
Ah didn't realize that method exists. Although looks like internally `FormBase.get_form()` fetches the app as well, plus it does another query on couch to check if the id is unique, so getting the app directly will be slightly faster.